frontend/src/components/Plans.tsx

```diff
@@ -74,8 +75,11 @@ export const Plans: React.FC = () => {
   }
   const [form, setForm] = React.useState<IPurchase>(getDefaults())
   const enterprise = !!license && !license.plan.billing
+  const userPlan = plans.find(plan => plan.id === license?.plan?.id) || plans[0]
   const personal = !license || license.plan.id === PERSONAL_PLAN_ID
-
+  const [confirm, setConfirm] = React.useState(false)
+  const [planId, setPlanId] = React.useState<string | undefined>(); 
+  const [priceId, setPriceId] = React.useState<string | undefined>(); 
```

Comment:
I think this can be done simpler by adding the `confirm` to the `IPurchase` type and form. Then onSelect check if downgrading and set `confirm` to true instead of `checkout` and after confirmation set `confirm` to false and `checkout` to true... does that make sense? Please let me know and we can discuss. But you should not need any of this new state or the new "userPlan". The plan is already set in the form by default.

_Thanks!_